lib/plugins/aws/package/compile/events/msk/index.js

```diff
@@ -0,0 +1,176 @@
+          type: 'number',
+          minimum: 1,
+          maximum: 10000,
+        },
+        enabled: {
+          type: 'boolean',
+        },
+        startingPosition: {
+          type: 'string',
+          enum: ['LATEST', 'TRIM_HORIZON'],
+        },
+        topic: {
+          type: 'string',
+        },
+      },
+      additionalProperties: false,
+      required: ['arn', 'topic'],
+    });
+  }
+
+  resolveDependsOn(funcRole) {
+    let dependsOn = 'IamRoleLambdaExecution';
+
+    if (funcRole) {
+      if (
+        // check whether the custom role is an ARN
+        typeof funcRole === 'string' &&
+        funcRole.indexOf(':') !== -1
+      ) {
+        dependsOn = [];
+      } else if (
+        // otherwise, check if we have an in-service reference to a role ARN
+        typeof funcRole === 'object' &&
+        'Fn::GetAtt' in funcRole &&
+        Array.isArray(funcRole['Fn::GetAtt']) &&
+        funcRole['Fn::GetAtt'].length === 2 &&
+        typeof funcRole['Fn::GetAtt'][0] === 'string' &&
+        typeof funcRole['Fn::GetAtt'][1] === 'string' &&
+        funcRole['Fn::GetAtt'][1] === 'Arn'
+      ) {
+        dependsOn = funcRole['Fn::GetAtt'][0];
+      } else if (
+        // otherwise, check if we have an import or parameters ref
+        typeof funcRole === 'object' &&
+        ('Fn::ImportValue' in funcRole || 'Ref' in funcRole)
+      ) {
+        dependsOn = [];
+      } else if (typeof funcRole === 'string') {
+        dependsOn = funcRole;
+      }
+    }
+
+    return dependsOn;
+  }
+
+  getMSKClusterName(eventSourceArn) {
+    if (eventSourceArn['Fn::ImportValue']) {
+      return eventSourceArn['Fn::ImportValue'];
+    } else if (eventSourceArn.Ref) {
+      return eventSourceArn.Ref;
+    }
+
+    return eventSourceArn.split('/')[1];
+  }
+
+  compileMSKEvents() {
+    this.serverless.service.getAllFunctions().forEach(functionName => {
+      const functionObj = this.serverless.service.getFunction(functionName);
+      const cfTemplate = this.serverless.service.provider.compiledCloudFormationTemplate;
+
+      if (functionObj.events) {
+        // It is required to add the following statement in order to be able to connect to MSK cluster
+        const ec2Statement = {
+          Effect: 'Allow',
+          Action: [
+            'ec2:CreateNetworkInterface',
+            'ec2:DescribeNetworkInterfaces',
+            'ec2:DescribeVpcs',
+            'ec2:DeleteNetworkInterface',
+            'ec2:DescribeSubnets',
+            'ec2:DescribeSecurityGroups',
+          ],
+          Resource: '*',
+        };
+        const mskStatement = {
+          Effect: 'Allow',
+          Action: ['kafka:DescribeCluster', 'kafka:GetBootstrapBrokers'],
+          Resource: [],
+        };
+
+        functionObj.events.forEach(event => {
+          if (event.msk) {
+            const EventSourceArn = event.msk.arn;
+            const Topic = event.msk.topic;
+            const BatchSize = event.msk.batchSize || 100;
```

Comment:
Let's not set _default_ for optional properties (I know we sometimes do that in old code, but we have also history of errors cause of doing that).

It's safe to pass it as `undefined` to resource definition, as `undefined` values are automatically stripped on JSON stringification